test/Exception.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
Don't use.

Reply:
This is only used in a test file.
Without it, we would have to implement `isException` for each exception we're checking.

Reply:
Use `Proxy`, inline the definition or define `assertIsJust`.